scripts/data-fill.go

```diff
@@ -139,15 +143,33 @@ func main() {
 	}
 
 	service := getMetricsServiceImpl(opts)
-	for i := 0; i < *n; i++ {
-		metric := generateMetrics(opts, seedData)
-		// TODO: add options to send metric via HTTP and print JSON to stdout
-		_, err := service.Create(*metric)
-		if err != nil {
-			log.Printf("Error creating record %d: %v\n", i+1, err)
-		} else {
-			fmt.Printf("Created record %d\n", i+1)
+	if *nInit > 0 || *nSecurity > 0 {
+		opts.metricsTypes = initMetrics
+		for i := 0; i < *nInit; i++ {
+			createRandomMetric(i, service, opts, seedData)
+		}
+
+		// security metrics contain both init and security fields
+		opts.metricsTypes = initMetrics | securityMetrics
+		for i := 0; i < *nSecurity; i++ {
+			createRandomMetric(i, service, opts, seedData)
 		}
+	} else {
+		// generate all types of metrics data (-1 is 0x1111...)
+		opts.metricsTypes = -1
+		for i := 0; i < *n; i++ {
+			createRandomMetric(i, service, opts, seedData)
+		}
+	}
+}
+
+func createRandomMetric(i int, service web.MetricsServiceInterface, opts *SeedOptions, seedData *SeedData) {
+	metric := generateMetrics(opts, seedData)
+	_, err := service.Create(*metric)
+	if err != nil {
+		log.Printf("Error creating record %d: %v\n", i+1, err)
+	} else {
+		fmt.Printf("Created record %d\n", i+1)
```

Comment:
should this be `log` instead of `fmt`?

Reply:
Nice catch, not sure how I managed to write this!